Source/ChocolateyGui/Services/ChocoSettingsSourceService.cs

```diff
@@ -0,0 +1,87 @@
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ChocolateyGui.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System.Threading.Tasks;
+    using ChocolateyGui.AsyncProcess;
+    using ChocolateyGui.Models;
+    using ChocolateyGui.Providers;
+    using ChocolateyGui.ViewModels.Items;
+
+    public class ChocoSettingsSourceService : ISourceService
+    {
+        private readonly IChocolateyConfigurationProvider chocolateyConfigurationProvider;
+        private string chocoExePath = string.Empty;
+
+        public ChocoSettingsSourceService(IChocolateyConfigurationProvider chocolateyConfigurationProvider)
+        {
+            this.chocolateyConfigurationProvider = chocolateyConfigurationProvider;
+            this.chocoExePath = Path.Combine(this.chocolateyConfigurationProvider.ChocolateyInstall, "choco.exe");
+        }
+
+        public event SourcesChangedEventHandler SourcesChanged;
+
+        public async void AddSource(SourceViewModel sourceViewModel)
+        {
+            if (sourceViewModel == null)
+            {
+                throw new ArgumentNullException("sourceViewModel");
+            }
+
+            var arguments = new StringBuilder();
+            arguments.AppendFormat("source add {0} {1} -y", sourceViewModel.Name, sourceViewModel.Url);
+
+            await ProcessEx.RunAsync(this.chocoExePath, arguments.ToString());
+
+            var sourcesChangedEvent = this.SourcesChanged;
+            if (sourcesChangedEvent != null)
+            {
+                sourcesChangedEvent(
+                    this,
+                    new SourcesChangedEventArgs(new List<SourceViewModel> { sourceViewModel }, new List<SourceViewModel>()));
+            }
+        }
+
+        public SourceViewModel GetDefaultSource()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public IEnumerable<SourceViewModel> GetSources()
+        {
+            var arguments = new StringBuilder();
+            arguments.AppendFormat("source list");
+
+            var results = ProcessEx.Run(this.chocoExePath, arguments.ToString());
+
+            var sources = new List<SourceViewModel>();
+
+            foreach (var source in results.StandardOutput)
+            {
+                var parts = source.Split('-');
+                if (parts.Count() == 2)
+                {
+                    sources.Add(new SourceViewModel { Name = parts[0].Trim(), Url = parts[1].Trim() });
+                }
+            }
+
+            return sources;
+        }
+
+        public void RemoveSource(SourceViewModel viewModel)
+        {
+            throw new System.NotImplementedException();

```

Comment:
Yes, we will.  As I mentioned, this is not a complete implementation.  Just putting it out there for conversation.  This was a quick spike that I did the other day, and I was mulling over it, when I saw what Brandon was working on, I thought I better do something with it.

What I have here is just enough to get the application to open :smile_cat: